src/pwm.rs

```diff
@@ -13,12 +13,24 @@
 
   ```rust
   let gpioa = ..; // Set up and split GPIOA
+  // Select the pins you want to use
   let pins = (
       gpioa.pa0.into_alternate_push_pull(&mut gpioa.crl),
       gpioa.pa1.into_alternate_push_pull(&mut gpioa.crl),
       gpioa.pa2.into_alternate_push_pull(&mut gpioa.crl),
       gpioa.pa3.into_alternate_push_pull(&mut gpioa.crl),
   );
+
+  // Set up the timer as a PWM output. Since there are multiple remap
+  // options for tim2 that use the same pins, you need to specify
+  // the remap generic parameter.
```

Comment:
cc: @burrbull 

In fact, this is not exactly so. It is possible to omit remap generic parameter in many cases when compiler is able to deduce mapping from the selected pins. It can be skipped on the first attempt and compiler will complain if it can not figure out the specific pin combination :)

How about updating this section, e.g. something along the following lines:

```
Set up the timer as a PWM output. If selected pins may correspond to different remap options,
then you shall specify the remap generic parameter. Otherwise, if there is no such ambiguity,
the remap generic parameter can be omitted without complains from the compiler.
```

Reply:
Yea, that's what I meant to say but I agree it's pretty unclear. Your wording is a lot better :)